flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/AggregationCodeGenerator.scala

```diff
@@ -405,6 +481,17 @@ class AggregationCodeGenerator(
       }
     }
 
+    val aggFuncCode = Seq(
+      genSetAggregationResults,
+      genAccumulate,
+      genRetract,
+      genCreateAccumulators,
+      genSetForwardedFields,
+      genSetConstantFlags,
+      genCreateOutputRow,
+      genMergeAccumulatorsPair,
+      genResetAccumulator).mkString("\n")
```

Comment:
I think we need an additional `cleanUp()` method that clears all state objects for the current key. Otherwise, we will have memory leaks. The `cleanUp()` method must be called when the state retention timers trigger.